scwidgets/widgets.py

```diff
@@ -0,0 +1,230 @@
+        name: name of widget to display
+        inputs (dict): dict of parameter_name: paramter_converter
+        descriptions (dict): dict of parameter_name: paramter_description.
+        If a parameter_name does not appear in the dict if will be used as
+        it's own description
+        options (dict): a dict of parameter_name: parameter_options if you wish
+        to display a list of options for a given parameter.
+        """
+        super().__init__()
+        self.scope = scope
+        self.callable = callable
+        self.inputs = inputs
+        self.input_widgets = []
+        self._setup_input_widgets(inputs, descriptions, options)
+
+        self.output = widgets.Text(placeholder='output name',
+                                   value='',
+                                   continuous_update=False)
+
+        self.button = widgets.Button(description=name)
+        self.button.on_click(self._on_button_clicked)
+
+        self.output_area = widgets.Output()
+
+        self.children = [
+            widgets.VBox([
+                widgets.HBox(self.input_widgets + [self.output, self.button]),
+                self.output_area
+            ])
+        ]
+
+        self.subscribers = []
+
+    def _setup_input_widgets(self, inputs, descriptions, options):
+        """
+        Creates a Combobox widget for each entry in the inputs dict.
+        The placeholder is set based on the descriptions dict and the
+        options are set based on the options dict.
+        """
+        for name in inputs.keys():
+            placeholder = descriptions[name] if name in descriptions else name
+            option = options[name] if name in options else []
+            option = option() if callable(option) else option
+            self.input_widgets.append(
+                widgets.Combobox(placeholder=placeholder,
+                                 continuous_update=False,
+                                 options=option))
+
+    def _retrive_kwargs(self):
+        kwargs = {
+            name: converter(item.value)
+            for name, converter, item in zip(
+                self.inputs.keys(), self.inputs.values(), self.input_widgets)
+        }
+        return kwargs
+
+    def _on_button_clicked(self, button):
+        self.output_area.clear_output()
+        with self.output_area:
+            if self.output.value:
+                output_name = self.output.value
+            else:
+                print('Invalid inputs: No output name specified')
+                return
+            self.scope[output_name] = self.process()
+
+    def process(self):
+        """
+        Calls the wrapped function using the
+        parameter values specified.
+
+        Returns: Output of the wrapped callable.
+        """
+        try:
+            kwargs = self._retrive_kwargs()
+        except ValueError as e:
+            print(f'Invalid inputs: {e}')
+            return
+
+        return self.callable(**kwargs)
+
+
+def _repr_html_(scope):
```

Comment:
Maybe remove this for now? This seems like an advanced feature?

Reply:
Will do. In the future we could just have the default way to do this to as displaying the scipp module in a cell before the plotting and not doing it within the widget itself.